arclightning_backend/src/main.rs

```diff
@@ -2,56 +2,101 @@ extern crate futures;
 extern crate hyper;
 #[macro_use]
 extern crate serde_derive;
-#[macro_use]
 extern crate serde_json;
 extern crate toml;
 
 use futures::future;
-use hyper::rt::{Future, Stream};
+use hyper::rt::Future;
 use hyper::service::service_fn;
 use hyper::{Body, Method, Request, Response, Server, StatusCode};
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{self, ErrorKind, Read};
 use std::path::PathBuf;
+use std::sync::{Arc, Mutex};
 
 // after this concept is further understood, will switch to 'Either'
 type ResponseFuture = Box<Future<Item = Response<Body>, Error = hyper::Error> + Send>;
 
-#[derive(Debug, Serialize, Deserialize, PartialEq)]
+// using PartialEq for unit tests
+// Using clone in a unit test atm.  Might not be necessary
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
 struct Game {
     name: String,
     description: String,
-    genre: Vec<String>,
+    genres: Vec<String>,
     thumbnail_path: PathBuf,
     exe_path: PathBuf,
 }
 
-fn router(request: Request<Body>) -> ResponseFuture {
-    let mut games_list: Vec<Game> = Vec::new();
-
+fn router(games_arc: &Arc<Mutex<HashMap<String, Game>>>, request: Request<Body>) -> ResponseFuture {
     let mut response = Response::new(Body::empty());
+    let mut response_tuple: (hyper::Body, hyper::StatusCode) =
+        (Body::empty(), StatusCode::NOT_FOUND);
```

Comment:
Is this default value required if we're already setting it in the fallback case on line 60?